openwisp_controller/config/base/device.py

```diff
@@ -174,20 +182,32 @@ def save(self, *args, **kwargs):
         update_fields = kwargs.get('update_fields')
         if not update_fields or 'name' in update_fields:
             self._check_name_changed()
+        if not update_fields or 'group' in update_fields:
+            self._check_group_changed()
```

Comment:
it looks to me that this will hit the DB twice, please test this using `assertNumQueries`, let's avoid hitting the DB multiple times for the same info.